src/basic/rlimit-util.c

```diff
@@ -43,6 +43,8 @@ int setrlimit_closest(int resource, const struct rlimit *rlim) {
             fixed.rlim_max == highest.rlim_max)
                 return 0;
 
+        log_warning("Failed at setting rlimit %" PRIu64 " for resource RLIMIT_%s. Will attempt setting value %" PRIu64 " instead.", rlim->rlim_max, rlimit_to_string(resource), fixed.rlim_max);
```

Comment:
This is a "library-like" function, so we usually log at log_debug at most. See: https://github.com/systemd/systemd/blob/main/docs/CODING_STYLE.md#logging

Reply:
also please use RLIM_FMT

Reply:
Both done - cheers for the heads up :+1:  

Is there any chance to still log at `warn` level here ? 
E.g. at the moment, if a limit a user sets on a service fails to be enforced for `EPERM` reason, it'll be silently lowered.

Reply:
No, unless the code was reworked to percolate the error up the call stack. I don't think it makes sense.

Reply:
Yep, by doing this commit I had the intention to bubble up the information to the user somehow.

If warn-logging here is not an option, what would be a good way to propagate that info do you think ? 
I was thinking about maybe fixing `systemctl show` so it shows the value actually set.

Reply:
some of our library-like calls support a flag or a log level passed in, how one can ask them to log about errors themselves above LOG_DEBUG. THis might be a candidate for the same logic.

Where exactly did you run into this though?